tests/chainer_tests/links_tests/connection_tests/test_convolution_2d.py

```diff
@@ -5,137 +5,106 @@
 
 import chainer
 from chainer.backends import cuda
-from chainer import gradient_check
 from chainer import links
 from chainer import testing
 from chainer.testing import attr
-from chainer.testing import condition
 from chainer.utils import conv
 
 
 @testing.parameterize(*testing.product({
     'x_dtype': [numpy.float16, numpy.float32, numpy.float64],
     'W_dtype': [numpy.float16, numpy.float32, numpy.float64],
 }))
-class TestConvolution2D(unittest.TestCase):
+@testing.inject_backend_tests(
+    ['test_forward', 'test_backward', 'test_double_backward',
+     'test_pickling'],
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'use_cudnn': ['never', 'always'],
+        'cuda_device': [0, 1],
+    })
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+class TestConvolution2D(testing.LinkTestCase):
+
+    param_names = ('W', 'b')
 
     def setUp(self):
-        self.link = links.Convolution2D(
-            3, 2, 3, stride=2, pad=1,
-            initialW=chainer.initializers.Normal(1, self.W_dtype),
-            initial_bias=chainer.initializers.Normal(1, self.x_dtype))
-        self.link.cleargrads()
-
-        self.x = numpy.random.uniform(-1, 1,
-                                      (2, 3, 4, 3)).astype(self.x_dtype)
-        self.gy = numpy.random.uniform(-1, 1,
-                                       (2, 2, 2, 2)).astype(self.x_dtype)
-        self.check_backward_options = {}
+        self.check_backward_options.update({
+            'atol': 5e-4, 'rtol': 5e-3
+        })
         if self.x_dtype == numpy.float16 or self.W_dtype == numpy.float16:
-            self.check_backward_options = {'atol': 3e-2, 'rtol': 5e-2}
+            self.check_forward_options.update({'atol': 5e-3, 'rtol': 5e-2})
+            self.check_backward_options.update({
+                'atol': 1e-3, 'rtol': 1e-3
```

Comment:
Tolerances are modified. Is it intentional?

Reply:
Yes, original test had retries and it fails a lot when they are removed.